packages/components/src/List/Toolbar/ColumnChooserButton/README.md

```diff
@@ -0,0 +1,249 @@
+
+| Props | Type | Info
+|---|---|---|
+| children | function | the body's  content of the popover, called with the columns as arguments.
+
+### ColumnChooserRowRenderer
+
+The row renderer only render its children.
+It holds reference to other components: , .
+It consumes the [columnChooserContext](#columnchoosercontext).
+
+The component holds reference :
+* Visibility: [RowVisibilityCheckbox](#rowvisibilitycheckbox)
+* Label: [RowLabel](#rowlabel)
+
+| Props | Type | Info
+|---|---|---|
+| children | React.element, [React.element] | the row's  content of the popover
+
+### RowVisibilityCheckbox
+
+Add a checkbox to switch the visibility of a column.
+If the column is locked, a locked icon replaces the checkbox.
+It consumes the [columnChooserContext](#columnchoosercontext).
+
+| Props | Type | Info
+|---|---|---|
+| index | number | the index of the column
+| locked | bool | indicates if the column is locked or not
+| value | bool | indicates the visibility value, and if the checkbox is checked
+
+### RowLabel
+
+Add the label of the column to the row.
+
+| Props | Type | Info
+|---|---|---|
+| label | string | the label of the column
+
+## Hooks
+### ColumnChooserManagerHook
+
+This hook manages all the logic behavior in the [ColumnChooser](#columnchooser).
+It holds the state of the columns edited by the user.
+
+| Parameters | Type | Info
+|---|---|---|
+| initColumns | array | array of columns object from the list
+| nbLockedLeftItems | number | the number of columns that will be locked, left to right
+
+And it returns an object with some values and handlers
+
+| Fields | Type | Info
+|---|---|---|
+| onChangeVisibility | function | change the visibility of the given column index|
+| onSelectAll | function | change the visibility of every columns
+| columnsChooser | [ColumnsChooserPropTypes](#columnschooserproptypes) | the hook columns
+| selectAll | bool | state of the select all button, checked or not.
+
+## Context
+
+### columnChooserContext
+
+The column chooser uses a context to pass some props to its children and avoir props drilling.
+By default the context is initialized in [ColumnChooser](#columnchooser).
+You can import directly the provider and the consumer from columnChooser.context
+```
+import { columnChooserContext, ColumnChooserProvider } 
+															from '.../columnChooserContext'
+```
+
+## PropTypes and Structures
+
+### ColumnChooserPropTypes
+
+| Props | Type | Info
+|---|---|---|
+| label | PropTypes.string | the label of the column
+| hidden | PropTypes.bool | the state of visibility of the column
+| order | PropTypes.number | indicates the ordering of the column
+
+### ColumnsChooserPropTypes
+
+An array of [ColumnChooserPropTypes](#columnchooserproptypes)
+
+## Utils
+### mergedColumnsChooser
+Helper to merge the columns chooser collection into the columns list.
+
+### compareOrder
+Compare the order of the columns chooser collection and order them ascending
```

Comment:
maybe add link to an implementation example

Reply:
We have the storybook for full example.